content/spin/kv-store-api-guide.md

```diff
@@ -0,0 +1,151 @@
+|------------|------------|---------|----------|
+| `open`  | name | store  | Open the store with the specified name. If `name` is the string "default", the default store is opened for the component that was granted access in the component manifest from `spin.toml`. Otherwise, `name` must refer to a store defined and configured in a [runtime configuration file](./dynamic-configuration.md#key-value-store-runtime-configuration) supplied with the application.|
+| `get` | store, key | key | Get the value associated with the specified `key` from the specified `store`. |
+| `set` | store, key, value | - | Set the `value` associated with the specified `key` in the specified `store`, overwriting any existing value. |
+| `delete` | store, key | - | Delete the tuple with the specified `key` from the specified `store`. `error::invalid-store` will be raised if `store` is not a valid handle to an open store.  No error is raised if a tuple did not previously exist for `key`.|
+| `exists` | store, key | boolean | Return whether a tuple exists for the specified `key` in the specified `store`.|
+| `get-keys` | store | list<keys> | Return a list of all the keys in the specified `store`. |
+| `close` | store | - | Close the specified `store`. |
+
+The exact detail of calling these operations from your application depends on your language:
+
+{{ tabs "sdk-type" }}
+
+{{ startTab "Rust"}}
+
+Key value functions are available in the `spin_sdk::key_value` module. The function names match the operations above. For example:
+
+
+```rust
+use anyhow::Result;
+use spin_sdk::{
+    http::{Request, Response},
+    http_component,
+    key_value::{Error, Store},
+};
+#[http_component]
+fn handle_request(req: Request) -> Result<Response> {
+    let store = Store::open_default()?;
+    store.set("mykey", "myvalyue")?;
+    let value = store.get("mykey")?;
+     Ok(http::Response::builder().status(200).body(Some(value.into()))?)
+}
+```
+
+{{ blockEnd }}
+
+{{ startTab "Typescript"}}
+
+With Typescript, the key value functions can be accessed after opening a store using either the `spinSdk.kv.opn` or the `spinSdk.kv.openDefault` methods which returns a handle to the store. For example:
+
+```javascript
+import { HandleRequest, HttpRequest, HttpResponse } from "@fermyon/spin-sdk"
+
+export const handleRequest: HandleRequest = async function (request: HttpRequest): Promise<HttpResponse> {
+    let store = spinSdk.kv.openDefault()
+    store.set("mykey", "myvalue")
+    return {
+            status: 200,
+            headers: {"content-type":"text/plain"},
+            body: store.get("mykey")
+    }
+}
+
+```
+
+**General Notes**
+- The spinSdk object is always available at runtime. Code checking and completion are available in TypeScript at design time if the module imports anything from the @fermyon/spin-sdk package. For example: 
+
+`set` **Operation**
+- The value argument is of the type `ArrayBuffer | string`
+
+`get` **Operation**
+- If a key does not exist, it returns null
```

Comment:
This isn't true for `js2wasm v0.4.0` which is the latest release. I just double-checked. It is planned to be a part of the next release.